File: backend/src/main/java/de/neuefische/backend/organization/model/Course.java
package de.neuefische.backend.organization.model;

import de.neuefische.backend.api.dto.CourseType;
import lombok.With;

import java.util.Objects;

@With
public record Course(
        String id,
        Long apiCourseId,
        String courseName,
        String courseContent,
        String courseDegree,
        String educationVoucher,
        CourseType courseType,
        Long apiOrganizationId
) {
    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;
        if (o == null || getClass() != o.getClass())
            return false;
        Course course = (Course) o;
        return Objects.equals(apiCourseId, course.apiCourseId);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(apiCourseId);
    }
}

File: frontend/src/types/Course.ts
import {CourseType} from "./CourseType.ts";

export type Course= {
    id:string,
    apiCourseId: number,
    courseName:string,
    courseContent:string,
    courseDegree:string,
    educationVoucher:string,
    courseType: CourseType,
    apiOrganizationId: number;
}